### ISSUE TITLE
cannot stat 'rhasspy-voltron/rhasspy-server-hermes/web/docs'

### ISSUE THREAD

**Issue by NullEnt1ty** [rhasspy/rhasspy#46]
## What happened?

`make install` failed with the following error message:
```
cp: cannot stat '/home/pi/rhasspy-voltron/rhasspy-server-hermes/web/docs': No such file or directory
make: *** [Makefile:153: install-rhasspy] Error 1
```

## Steps to reproduce

1. Make sure you have a clean repository (that goes for your submodules too!)
   ```sh
   # Heads up: The following commands will DELETE ALL UNCOMMITTED CHANGES!
   git clean -fdx && git reset --hard
   git submodule foreach git clean -fdx
   git submodule foreach git reset --hard
   ```
2. `./configure --enable-in-place RHASSPY_LANGUAGE=de`
3. `make`
4. `make install`

## My environment

Device: Raspberry Pi 3B+
OS: Raspbian 10 (buster)
Kernel: Linux 4.19.118-v7+
HEAD at: 8e9943c8f502ac3939ce1c2fb53c6d0243ab0eeb

<details> 
  <summary>My generated Makefile</summary>

```
# Makefile.in
#
# Copyright © 2012, 2013 Brandon Invergo <brandon@invergo.net>
#
# Copying and distribution of this file, with or without modification,
# are permitted in any medium without royalty provided the copyright
# notice and this notice are preserved.  This file is offered as-is,
# without any warranty.

# List whatever files you want to include in your source distribution here.
# You can include whole directories but note that *everything* under that
# directory will be included
DISTFILES = PKG-INFO Makefile.in configure setup.py install-sh

DESTDIR =
PACKAGE_BUGREPORT = mike@rhasspy.org
PACKAGE_NAME = rhasspy
PACKAGE_STRING = rhasspy 2.5.0
PACKAGE_TARNAME = rhasspy
PACKAGE_DISTNAME = ${PACKAGE_NAME}-${PACKAGE_VERSION}
PACKAGE_URL =
PACKAGE_VERSION = 2.5.0
PATH_SEPARATOR = :
PYTHON = /usr/bin/python3
VIRTUALENV = yes
SHELL = /bin/bash
MKDIR_P = /bin/mkdir -p
INSTALL = /usr/bin/install -c
INSTALL_PROGRAM = ${INSTALL}
INSTALL_DATA = ${INSTALL} -m 644
INSTALL_SCRIPT = ${INSTALL}
exec_prefix = ${prefix}
includedir = ${prefix}/include
infodir = ${datarootdir}/info
prefix = /home/pi/rhasspy-voltron/.venv
srcdir = /home/pi/rhasspy-voltron
datadir = ${datarootdir}
datarootdir = ${prefix}/share
pkgdatadir = $(datadir)/rhasspy
pkgincludedir = $(includedir)/rhasspy

# -----------------------------------------------------------------------------
# Dependencies
# -----------------------------------------------------------------------------

BUILD_DIR ?= $(srcdir)/build
DOWNLOAD_DIR ?= $(srcdir)/download
DO_DOWNLOAD ?= curl -sSfL -o

DOCKER_ARCH = armv7

ENABLE_OPENGRM = yes
OPENGRM_ARCH = @OPENGRM_ARCH@
OPENGRM_FROM = prebuilt
OPENGRM_PRE_FILE = $(DOWNLOAD_DIR)/opengrm-1.3.4_$(DOCKER_ARCH).tar.gz
OPENGRM_SRC_FILE = $(DOWNLOAD_DIR)/opengrm-ngram-1.3.4.tar.gz
OPENFST_SRC_FILE = $(DOWNLOAD_DIR)/openfst-1.6.9.tar.gz

ENABLE_PHONETISAURUS = yes
PHONETISAURUS_FROM = prebuilt
PHONETISAURUS_PRE_FILE = $(DOWNLOAD_DIR)/phonetisaurus-2019_$(DOCKER_ARCH).tar.gz
PHONETISAURUS_SRC_FILE = $(DOWNLOAD_DIR)/phonetisaurus-2019.tar.gz

ENABLE_PRECISE = yes
PRECISE_ARCH = armv7l
PRECISE_FROM = prebuilt
PRECISE_FILE = $(DOWNLOAD_DIR)/precise-engine_0.3.0_$(DOCKER_ARCH).tar.gz

ENABLE_POCKETSPHINX = yes
POCKETSPHINX_FILE = $(DOWNLOAD_DIR)/pocketsphinx-python.tar.gz

ENABLE_KALDI = yes
KALDI_FROM = prebuilt
KALDI_PRE_FILE = $(DOWNLOAD_DIR)/kaldi-2020_$(DOCKER_ARCH).tar.gz
KALDI_SRC_FILE = $(DOWNLOAD_DIR)/kaldi-2020.tar.gz

ENABLE_JULIUS = no
JULIUS_FROM = prebuilt
JULIUS_PRE_FILE = $(DOWNLOAD_DIR)/julius-4.5_$(DOCKER_ARCH).tar.gz
JULIUS_SRC_FILE = $(DOWNLOAD_DIR)/julius-2019.tar.gz

ENABLE_DEEPSPEECH = no
DEEPSPEECH_ARCH = rpi3
DEEPSPEECH_FROM = prebuilt
DEEPSPEECH_FILE = $(DOWNLOAD_DIR)/native_client.$(DOCKER_ARCH).cpu.linux.0.6.1.tar.xz

ENABLE_KENLM = no
KENLM_FROM = prebuilt
KENLM_PRE_FILE = $(DOWNLOAD_DIR)/kenlm-20200308_$(DOCKER_ARCH).tar.gz
EIGEN_SRC_FILE = $(DOWNLOAD_DIR)/eigen-3.2.8.tar.bz2
KENLM_SRC_FILE = $(DOWNLOAD_DIR)/kenlm-20200308.tar.gz

ENABLE_NANOTTS = yes
NANOTTS_FROM = prebuilt
NANOTTS_PRE_FILE = $(DOWNLOAD_DIR)/nanotts-20200520_$(DOCKER_ARCH).tar.gz
NANOTTS_SRC_FILE = $(DOWNLOAD_DIR)/nanotts-20200520.tar.gz

ENABLE_SNOWBOY = yes
SNOWBOY_FILE = $(DOWNLOAD_DIR)/snowboy-1.3.0.tar.gz

# -----------------------------------------------------------------------------

VENV_DIR = $(DESTDIR)$(prefix)
VENV_PYTHON = $(VENV_DIR)/bin/python
VIRTUALENV_FLAGS ?= --clear

ifeq ($(VIRTUALENV),yes)
  INSTALL_PYTHON=$(VENV_PYTHON)
else # VIRTUALENV == yes
  INSTALL_PYTHON=$(PYTHON)
endif # VIRTUALENV == yes

.PHONY: all install uninstall distclean clean dist check installdirs

all: download-init build-init opengrm phonetisaurus precise pocketsphinx kaldi julius snowboy deepspeech kenlm

.PHONY: install-init install-dependencies install-rhasspy
install: installdirs \
         install-init \
         install-dependencies \
         install-rhasspy

install-init:
ifeq ($(VIRTUALENV),yes)
        "$(PYTHON)" -m venv $(VIRTUALENV_FLAGS) "$(VENV_DIR)" || "$(PYTHON)" -m venv "$(VENV_DIR)"
        "$(VENV_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) --upgrade pip
        "$(VENV_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) wheel || true
        "$(VENV_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) setuptools
endif # VIRTUALENV == yes

.PHONY: install-opengrm install-phonetisaurus install-precise install-pocketsphinx install-kaldi install-julius install-snowboy install-deepspeech install-kenlm
install-dependencies: install-opengrm \
                      install-phonetisaurus \
                      install-precise \
                      install-pocketsphinx \
                      install-kaldi \
                      install-julius \
                      install-snowboy \
                      install-deepspeech \
                      install-kenlm \
                      install-nanotts

install-rhasspy:
        "$(INSTALL_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) "$(srcdir)"
        chmod +x "${srcdir}/rhasspy.sh"
        install -D "${srcdir}/rhasspy.sh" "$(DESTDIR)$(prefix)/bin/rhasspy"
        install -D "--target-directory=$(DESTDIR)$(prefix)/bin" "${srcdir}/bin/rhasspy-voltron" "${srcdir}/bin/voltron-run"
        install -D "--target-directory=$(DESTDIR)$(prefix)/share/rhasspy/etc" "${srcdir}/etc/shflags"
        install -D "--target-directory=$(DESTDIR)$(prefix)/share/rhasspy/wav" "${srcdir}/etc/wav"/*.wav
        install -D "--target-directory=$(DESTDIR)$(prefix)/share/rhasspy" VERSION README.md LICENSE
        cp -fR "${srcdir}/rhasspy-server-hermes/web" "$(DESTDIR)$(prefix)/share/rhasspy/"
        cp -fR "${srcdir}/rhasspy-server-hermes/templates" "$(DESTDIR)$(prefix)/share/rhasspy/"
        cp -fR "${srcdir}/rhasspy-server-hermes/web/docs" "$(DESTDIR)$(prefix)/share/rhasspy/"

# -----------------------------------------------------------------------------

.PHONY: download-init
download-init:
        $(MKDIR_P) "$(DOWNLOAD_DIR)"

.PHONY: build-init
build-init:
        $(MKDIR_P) "$(BUILD_DIR)"

# -----------------------------------------------------------------------------
# Opengrm: languge modeling toolkit
# http://www.opengrm.org/twiki/bin/view/GRM/NGramLibrary
# -----------------------------------------------------------------------------

.PHONY: opengrm
ifeq ($(ENABLE_OPENGRM),yes)
opengrm: $(OPENGRM_PRE_FILE)

install-opengrm: $(OPENGRM_PRE_FILE)
        "$(srcdir)/scripts/install/install-opengrm.sh" "$(OPENGRM_PRE_FILE)" "$(DESTDIR)$(prefix)"

  ifeq ($(OPENGRM_FROM),prebuilt)
$(OPENGRM_PRE_FILE):
        $(DO_DOWNLOAD) "$(OPENGRM_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/opengrm-1.3.4_$(DOCKER_ARCH).tar.gz"
  else # OPENGRM_FROM
$(OPENGRM_PRE_FILE): $(OPENFST_SRC_FILE) $(OPENGRM_SRC_FILE)
        "$(srcdir)/scripts/build/build-opengrm.sh" "$(OPENGRM_SRC_FILE)" "$(OPENFST_SRC_FILE)" "$(BUILD_DIR)" "$(OPENGRM_PRE_FILE)"
  endif # OPENGRM_FROM
else # ENABLE_OPENGRM
opengrm:

install-opengrm:
endif # ENABLE_OPENGRM

$(OPENFST_SRC_FILE):
        $(DO_DOWNLOAD) "$(OPENFST_SRC_FILE)" 'http://www.openfst.org/twiki/pub/FST/FstDownload/openfst-1.6.9.tar.gz'

$(OPENGRM_SRC_FILE):
        $(DO_DOWNLOAD) "$(OPENGRM_SRC_FILE)" 'http://www.opengrm.org/twiki/pub/GRM/NGramDownload/opengrm-ngram-1.3.4.tar.gz'

# -----------------------------------------------------------------------------
# Phonetisaurus: grapheme to phoneme
# https://github.com/AdolfVonKleist/Phonetisaurus
# -----------------------------------------------------------------------------

.PHONY: phonetisaurus
ifeq ($(ENABLE_PHONETISAURUS),yes)
phonetisaurus: $(PHONETISAURUS_PRE_FILE)

install-phonetisaurus: $(PHONETISAURUS_PRE_FILE)
        "$(srcdir)/scripts/install/install-phonetisaurus.sh" "$(PHONETISAURUS_PRE_FILE)" "$(DESTDIR)$(prefix)"

  ifeq ($(PHONETISAURUS_FROM),prebuilt)
$(PHONETISAURUS_PRE_FILE):
        $(DO_DOWNLOAD) "$(PHONETISAURUS_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/phonetisaurus-2019_$(DOCKER_ARCH).tar.gz"
  else # PHONETISAURUS_FROM
$(PHONETISAURUS_PRE_FILE): $(OPENFST_SRC_FILE) $(PHONETISAURUS_SRC_FILE)
        "$(srcdir)/scripts/build/build-phonetisaurus.sh" "$(PHONETISAURUS_SRC_FILE)" "$(OPENFST_SRC_FILE)" "$(BUILD_DIR)" "$(PHONETISAURUS_PRE_FILE)"
  endif # PHONETISAURUS_FROM
else # ENABLE_PHONETISAURUS
phonetisaurus:

install-phonetisaurus:
endif # ENABLE_PHONETISAURUS

$(PHONETISAURUS_SRC_FILE):
        $(DO_DOWNLOAD) "$(PHONETISAURUS_SRC_FILE)" 'https://github.com/synesthesiam/prebuilt-apps/raw/master/download/phonetisaurus-2019.tar.gz'

# -----------------------------------------------------------------------------
# Mycroft Precise: wake word detection
# https://github.com/MycroftAI/mycroft-precise
# -----------------------------------------------------------------------------

.PHONY: precise install-precise
ifeq ($(ENABLE_PRECISE),yes)
precise: $(PRECISE_FILE)

install-precise: $(PRECISE_FILE)
        "$(srcdir)/scripts/install/install-precise.sh" "$(PRECISE_FILE)" "$(DESTDIR)$(prefix)"
else # ENABLE_PRECISE
precise:

install-precise:
endif # ENABLE_PRECISE

$(PRECISE_FILE):
        $(DO_DOWNLOAD) "$(PRECISE_FILE)" "https://github.com/MycroftAI/mycroft-precise/releases/download/v0.3.0/precise-engine_0.3.0_$(PRECISE_ARCH).tar.gz"

# -----------------------------------------------------------------------------
# Snowboy: wake word detection
# https://snowboy.kitt.ai
# -----------------------------------------------------------------------------

.PHONY: snowboy install-snowboy
ifeq ($(ENABLE_SNOWBOY),yes)
snowboy: $(SNOWBOY_FILE)

install-snowboy: $(SNOWBOY_FILE)
        "$(INSTALL_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) "$(SNOWBOY_FILE)"
else # ENABLE_SNOWBOY
snowboy:

install-snowboy:
endif # ENABLE_SNOWBOY

$(SNOWBOY_FILE):
        $(DO_DOWNLOAD) "$(SNOWBOY_FILE)" 'https://github.com/Kitt-AI/snowboy/archive/v1.3.0.tar.gz'

# -----------------------------------------------------------------------------
# CMU Pocketsphinx: speech to text
# https://github.com/cmusphinx/pocketsphinx
# -----------------------------------------------------------------------------

.PHONY: pocketsphinx install-pocketsphinx
ifeq ($(ENABLE_POCKETSPHINX),yes)
pocketsphinx: $(POCKETSPHINX_FILE)

install-pocketsphinx: $(POCKETSPHINX_FILE)
        "$(INSTALL_PYTHON)" -m pip install $(PIP_INSTALL_ARGS) "$(POCKETSPHINX_FILE)"
else # ENABLE_POCKETSPHINX
pocketsphinx:

install-pocketsphinx:
endif # ENABLE_POCKETSPHINX

$(POCKETSPHINX_FILE):
        $(DO_DOWNLOAD) "$(POCKETSPHINX_FILE)" 'https://github.com/synesthesiam/pocketsphinx-python/releases/download/v1.0/pocketsphinx-python.tar.gz'

# -----------------------------------------------------------------------------
# Kaldi: speech to text
# https://kaldi-asr.org
# -----------------------------------------------------------------------------

.PHONY: kaldi
ifeq ($(ENABLE_KALDI),yes)
kaldi: $(KALDI_PRE_FILE)

install-kaldi: $(KALDI_PRE_FILE)
        "$(srcdir)/scripts/install/install-kaldi.sh" "$(KALDI_PRE_FILE)" "$(DESTDIR)$(prefix)"

  ifeq ($(KALDI_FROM),prebuilt)
$(KALDI_PRE_FILE):
        $(DO_DOWNLOAD) "$(KALDI_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/kaldi-2020_$(DOCKER_ARCH).tar.gz"
  else # KALDI_FROM
$(KALDI_PRE_FILE): $(KALDI_SRC_FILE)
        "$(srcdir)/scripts/build/build-kaldi.sh" "$(KALDI_SRC_FILE)" "$(BUILD_DIR)" "$(KALDI_PRE_FILE)"
  endif # KALDI_FROM
else # ENABLE_KALDI
kaldi:

install-kaldi:
endif # ENABLE_KALDI

$(KALDI_SRC_FILE):
        $(DO_DOWNLOAD) "$(KALDI_SRC_FILE)" 'https://github.com/synesthesiam/docker-kaldi/raw/master/download/kaldi-2020.tar.gz'

# -----------------------------------------------------------------------------
# Julius: speech to text
# https://github.com/julius-speech/julius
# -----------------------------------------------------------------------------

.PHONY: julius
ifeq ($(ENABLE_JULIUS),yes)
julius: $(JULIUS_PRE_FILE)

install-julius: $(JULIUS_PRE_FILE)
        "$(srcdir)/scripts/install/install-julius.sh" "$(JULIUS_PRE_FILE)" "$(DESTDIR)$(prefix)"

  ifeq ($(JULIUS_FROM),prebuilt)
$(JULIUS_PRE_FILE):
        $(DO_DOWNLOAD) "$(JULIUS_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/julius-4.5_$(DOCKER_ARCH).tar.gz"
  else # JULIUS_FROM
$(JULIUS_PRE_FILE): $(JULIUS_SRC_FILE)
        "$(srcdir)/scripts/build/build-julius.sh" "$(JULIUS_SRC_FILE)" "$(BUILD_DIR)" "$(JULIUS_PRE_FILE)"
  endif # JULIUS_FROM
else # ENABLE_JULIUS
julius:

install-julius:
endif # ENABLE_JULIUS

$(JULIUS_SRC_FILE):
        $(DO_DOWNLOAD) "$(JULIUS_SRC_FILE)" 'https://github.com/synesthesiam/prebuilt-apps/raw/master/download/julius-2019.tar.gz'


# -----------------------------------------------------------------------------
# Mozilla DeepSpeech: speech to text
# https://github.com/mozilla/DeepSpeech
# -----------------------------------------------------------------------------

.PHONY: deepspeech install-deepspeech
ifeq ($(ENABLE_DEEPSPEECH),yes)
deepspeech: $(DEEPSPEECH_FILE)

install-deepspeech: $(DEEPSPEECH_FILE)
        "$(srcdir)/scripts/install/install-deepspeech.sh" "$(DEEPSPEECH_FILE)" "$(DESTDIR)$(prefix)"
else # ENABLE_DEEPSPEECH
deepspeech:

install-deepspeech:
endif # ENABLE_DEEPSPEECH

$(DEEPSPEECH_FILE):
        $(DO_DOWNLOAD) "$(DEEPSPEECH_FILE)" "https://github.com/mozilla/DeepSpeech/releases/download/v0.6.1/native_client.$(DEEPSPEECH_ARCH).cpu.linux.tar.xz"

# -----------------------------------------------------------------------------
# KenLM: language modeling toolkit
# https://kheafield.com/code/kenlm/
# -----------------------------------------------------------------------------

.PHONY: kenlm install-kenlm
ifeq ($(ENABLE_KENLM),yes)
kenlm: $(KENLM_PRE_FILE)

  ifeq ($(KENLM_FROM),prebuilt)
$(KENLM_PRE_FILE):
        $(DO_DOWNLOAD) "$(KENLM_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/kenlm-20200308_$(DOCKER_ARCH).tar.gz"
  else # KENLM_FROM
$(KENLM_PRE_FILE): $(EIGEN_SRC_FILE) $(KENLM_SRC_FILE)
        "$(srcdir)/scripts/build/build-kenlm.sh" "$(KENLM_SRC_FILE)" "$(EIGEN_SRC_FILE)" "$(BUILD_DIR)" "$(KENLM_PRE_FILE)"
  endif # KENLM_FROM

install-kenlm: $(KENLM_PRE_FILE)
        "$(srcdir)/scripts/install/install-kenlm.sh" "$(KENLM_PRE_FILE)" "$(DESTDIR)$(prefix)"
else # ENABLE_KENLM
kenlm:

install-kenlm:
endif # ENABLE_KENLM

$(KENLM_SRC_FILE):
        $(DO_DOWNLOAD) "$(KENLM_SRC_FILE)" 'https://github.com/synesthesiam/prebuilt-apps/raw/master/download/kenlm-20200308.tar.gz'

$(EIGEN_SRC_FILE):
        $(DO_DOWNLOAD) "$(EIGEN_SRC_FILE)" 'https://github.com/synesthesiam/prebuilt-apps/raw/master/download/eigen-3.2.8.tar.bz2'

# -----------------------------------------------------------------------------
# nanoTTS: text to speech
# https://github.com/gmn/nanotts
# -----------------------------------------------------------------------------

.PHONY: nanotts
ifeq ($(ENABLE_NANOTTS),yes)
nanotts: $(NANOTTS_PRE_FILE)

install-nanotts: $(NANOTTS_PRE_FILE)
        "$(srcdir)/scripts/install/install-nanotts.sh" "$(NANOTTS_PRE_FILE)" "$(DESTDIR)$(prefix)"

  ifeq ($(NANOTTS_FROM),prebuilt)
$(NANOTTS_PRE_FILE):
        $(DO_DOWNLOAD) "$(NANOTTS_PRE_FILE)" "https://github.com/synesthesiam/prebuilt-apps/releases/download/v1.0/nanotts-20200520_$(DOCKER_ARCH).tar.gz"
  else # NANOTTS_FROM
$(NANOTTS_PRE_FILE): $(NANOTTS_SRC_FILE)
        "$(srcdir)/scripts/build/build-nanotts.sh" "$(NANOTTS_SRC_FILE)" "$(BUILD_DIR)" "$(NANOTTS_PRE_FILE)"
  endif # NANOTTS_FROM
else # ENABLE_NANOTTS
nanotts:

install-nanotts:
endif # ENABLE_NANOTTS

$(NANOTTS_SRC_FILE):
        $(DO_DOWNLOAD) "$(NANOTTS_SRC_FILE)" 'https://github.com/synesthesiam/prebuilt-apps/raw/master/download/nanotts-20200520.tar.gz'

# -----------------------------------------------------------------------------

uninstall:
ifeq ($(VIRTUALENV),yes)
        rm -rvf "$(VENV_DIR)"
endif


# Delete downloaded binaries and build artifacts
clean:
        "$(PYTHON)" setup.py clean
        rm -rvf "$(BUILD_DIR)"


# Clean up the output of configure
distclean:
        rm -v "$(srcdir)/config.log"
        rm -v "$(srcdir)/config.status"
        rm -rvf "$(srcdir)/autom4te.cache"
        rm -v "$(srcdir)/Makefile"

# You can either use the setup.py sdist command or you can roll your own here
dist:
        "$(PYTHON)" setup.py sdist

# Use the setup.py check command
check:
        "$(PYTHON)" setup.py check

# setup.py might complain if a directory doesn't exist so just in case, make the directory
# here
installdirs:
        $(MKDIR_P) "$(DESTDIR)$(prefix)"
```
</details>

**Comment by synesthesiam** [comment]
Sorry about that. For now, you can delete the last line of the `install-rhasspy` target in your generated Makefile (the one that references `rhasspy-server-hermes/web/docs`).

Building the docs requires `mkdocs`, which is not installed by default in the virtual environment (it pulls in a lot of extra unnecessary dependencies, in my opinion). I'm still trying to figure out how best to do this.

**Comment by snail59** [comment]
Hello everyone reading this and @synesthesiam,

I was facing this problem and  removed the line you wrote about. Then I found this bug report.

I am not a developer so I may be wrong, but the problem seems to be that the rhasspy-server-hermes/web/docs directory has been removed ( commit e9ddf055d10c643988d4166157ec086c57ee2324 ) while this directory is still referenced in the Makefile.in of rhasspy-voltron.

I installed mkdocs and still, neither the files nor the directory are created.
